Add dynamic data table on collection page

ml-evs opened the pull request:
Closes #815
Closes #891.

Replaces the next table on collections with the new dynamic table, and:

- [x] Customises the edit page links on a per-table basis to allow it to work with the current collections routes (closes #891)
- [x] Allows you to disable the buttons and top search bar

BenjaminCharmes commented on webapp/src/components/CollectionTable.vue:
```diff
@@ -4,6 +4,7 @@
     :data="collections"
     :data-type="'collections'"
     :global-filter-fields="['collection_id', 'title', 'creatorsList']"
+    :edit-page-route-prefix="collections"
```
Should be a string ?

BenjaminCharmes replied:
Should be a string ?

```suggestion
    :edit-page-route-prefix="'collections'"
```

BenjaminCharmes replied:
This should be a string and not an object ?

BenjaminCharmes commented on webapp/src/components/DynamicDataTable.vue:
```diff
@@ -222,9 +228,9 @@ export default {
         event.originalEvent.altKey
       ) {
-        window.open(`/edit/${row.item_id}`, "_blank");
+        window.open(`/${this.editPageRoutePrefix}/${row.item_id}`, "_blank");
```
Missing in props ?

```suggestion
props: {
  editPageRoutePrefix: {
    type: String,
    required: true,
  },
},```

BenjaminCharmes commented on webapp/src/components/DynamicDataTable.vue:
```diff
@@ -222,9 +228,9 @@ export default {
         event.originalEvent.metaKey ||
         event.originalEvent.altKey
       ) {
-        window.open(`/edit/${row.item_id}`, "_blank");
```
`editPageRoutePrefix` is missing inside `Props` ?

BenjaminCharmes reviewed (commented):
Looks really good. Just a few comments or things to modify maybe?

ml-evs commented:
Thanks Ben, think I've fixed everything

> Perhaps we should either remove the checkboxes column inside CollectionInformation, or add a button to delete multiple items from the collection?

I've removed the checkboxes for now, though we might think about a delete button that removes from just a collection, good idea!

> And a small eslint warning inside `CollectionInformation.vue` because "data" should be above "computed".

Just raised #895 for this

BenjaminCharmes reviewed (approved):
Very great ! :rocket:

ml-evs commented:
Think this is ready again when you are @BenjaminCharmes! I added a test to catch the case we discussed (also our first for the collections table/page generally).

BenjaminCharmes reviewed (approved):
It look really great, nice one ! 🚀

ml-evs commented:
> It look really great, nice one ! 🚀

Thanks, I'm going to merge this now but make a mental note that the tests locally might not work every time
